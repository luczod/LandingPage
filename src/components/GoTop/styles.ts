import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    position: fixed;
    right: 3rem;
    bottom: 2rem;
    z-index: 6;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 4rem;
    width: 4rem;
    opacity: 0.8;
  `}
`;
