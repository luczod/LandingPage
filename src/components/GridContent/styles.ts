import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: 0;
    margin: 0 auto;
  `}
`;

export const HtmlWrap = styled.div`
  ${({ theme }) => css`
    margin: 0;
  `}
`;
