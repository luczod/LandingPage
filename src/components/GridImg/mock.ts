export default {
  bg: false,
  title: "My image grid",
  description:
    "Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde",
  img_grid: [
    {
      id: 8,
      image: {
        data: [
          {
            id: 8,
            attributes: {
              name: "Teste 1",
              alternativeText: "Teste 1",
              url: "https://source.unsplash.com/random/800x800?r=1",
            },
          },
        ],
      },
    },
    {
      id: 9,
      image: {
        data: [
          {
            id: 9,
            attributes: {
              name: "Teste 2",
              alternativeText: "Teste 2",
              url: "https://source.unsplash.com/random/800x800?r=2",
            },
          },
        ],
      },
    },
    {
      id: 10,
      image: {
        data: [
          {
            id: 10,
            attributes: {
              name: "Teste 3",
              alternativeText: "Teste 3",
              url: "https://source.unsplash.com/random/800x800?r=3",
            },
          },
        ],
      },
    },
  ],
};
