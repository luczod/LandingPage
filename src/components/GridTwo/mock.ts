export default {
  title: "Grid two columns",
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Perspiciatis harum odio tenetur consequatur obcaecati, earum perferendis commodi id repellat beatae, maiores numquam quos enim necessitatibus ipsam fugiat, incidunt consectetur aperiam!`,
  srcImg: "assets/images/javascript.svg",
  bg: true,
};
