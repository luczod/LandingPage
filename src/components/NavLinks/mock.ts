export const linksList = [
  {
    id: "1",
    link_text: "url 1",
    url: "#target1",
    open_in_new_tab: false,
  },
  {
    id: "2",
    link_text: "url 2",
    url: "#target2",
    newTab: false,
  },
  {
    id: "3",
    link_text: "url 3",
    url: "#target3",
    newTab: false,
  },
  {
    id: "4",
    link_text: "url 4",
    url: "#target4",
    newTab: false,
  },
  {
    id: "5",
    link_text: "url 5",
    url: "#target5",
    newTab: false,
  },
  {
    id: "6",
    link_text: "url 6",
    url: "#target6",
    newTab: false,
  },
  {
    id: "7",
    link_text: "url 7",
    url: "#target7",
    newTab: false,
  },
  {
    id: "8",
    link_text: "url 8",
    url: "#target8",
    newTab: false,
  },
  {
    id: "9",
    link_text: "url 9",
    url: "#target9",
    newTab: false,
  },
  {
    id: "10",
    link_text: "url 10",
    url: "#target10",
    newTab: false,
  },
];
