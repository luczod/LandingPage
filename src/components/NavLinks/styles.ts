import styled, { css } from "styled-components";

export const Container = styled.nav`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    display: flex;
    /* flex direction and flex-wrap */
    flex-flow: row wrap;

    @media ${theme.media.lteMedium} {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
