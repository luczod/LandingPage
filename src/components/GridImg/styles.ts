import styled, { css } from "styled-components";
import { Container as TextComponent } from "../Textfn/styles";

interface IPros {
  open?: boolean;
}

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
  `}
`;

export const GridItem = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 1rem;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;

export const ModalImg = styled.div`
  ${({ theme }) => css`
    display: none; /* Hidden by default */
    position: fixed; /* Stay in place */
    z-index: 1; /* Sit on top */
    padding-top: 100px; /* Location of the box */
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0 0 0 / 0.3); /* color-Black w/ opacity */
    border-radius: 1rem;

    & > span {
      position: fixed;
      cursor: pointer;
      top: 4rem;
      right: 2rem;
      color: ${theme.colors.secondaryColor};
      font-size: ${theme.font.sizes.hg};
      &::hover {
        color: ${theme.colors.white};
      }
      @media only screen and (max-width: 768px) {
        left: 1rem;
      }
    }

    & > img {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      border-radius: 1rem;
      /* 100% Image Width on Smaller Screens */
      @media only screen and (max-width: 768px) {
        width: 100%;
        padding: 0 0.8rem;
      }
    }
    & > div {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      text-align: center;
      font-size: ${theme.font.sizes.md};
      color: ${theme.colors.white};
      padding: 10px 0;
      height: 150px;
    }
  `}
`;
