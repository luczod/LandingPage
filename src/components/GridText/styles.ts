import styled, { css } from "styled-components";
import { Title as ItemTitle } from "../Heading/styles";
import { Container as TextComponent } from "../Textfn/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    overflow: hidden;
    width: 100%;
    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridItem = styled.article`
  ${({ theme }) => css`
    ${ItemTitle} {
      position: relative;
      left: 5rem;
    }

    ${ItemTitle}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhg};
      top: -2rem;
      left: -5rem;
      transform: rotate(8deg);
    }
  `}
`;
