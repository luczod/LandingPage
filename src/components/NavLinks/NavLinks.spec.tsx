import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { NavLinks } from ".";
import { linksList } from "./mock";
import { themefn } from "../../styles/themefn";
//jest -t NavLinks name-of-spec

describe("<NavLinks/>", () => {
  it("should render links", () => {
    renderTheme(<NavLinks Links={linksList} />);
    expect(screen.getAllByRole("link")).toHaveLength(linksList.length);
  });

  it("should not render links", () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/link/i)).toHaveLength(0);
  });
  /*   it("should render links media", () => {
    const { container } = renderTheme(<NavLinks Links={linksList} />);

    expect(container).toHaveStyleRule("flex-flow", "column wrap", {
      media: "(max-width: 660px)",
    });
    screen.debug();
  }); */

  it("should Match snapshot", () => {
    renderTheme(<NavLinks Links={linksList} />);
    expect(screen.getAllByRole("link")).toMatchSnapshot();
  });
  it("parentElement should Match snapshot", () => {
    const { container } = renderTheme(<NavLinks Links={linksList} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
