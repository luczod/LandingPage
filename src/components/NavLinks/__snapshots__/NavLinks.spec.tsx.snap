// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks/> parentElement should Match snapshot 1`] = `
.c0 {
  color: #FFFFFF;
  display: flex;
  flex-flow: row wrap;
}

.c1 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c1::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {
  .c0 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="#target1"
    target="_self"
  >
    Link 1
  </a>
  <a
    class="c1"
    href="#target2"
    target="_self"
  >
    Link 2
  </a>
  <a
    class="c1"
    href="#target3"
    target="_self"
  >
    Link 3
  </a>
  <a
    class="c1"
    href="#target4"
    target="_self"
  >
    Link 4
  </a>
  <a
    class="c1"
    href="#target5"
    target="_self"
  >
    Link 5
  </a>
  <a
    class="c1"
    href="#target6"
    target="_self"
  >
    Link 6
  </a>
  <a
    class="c1"
    href="#target7"
    target="_self"
  >
    Link 7
  </a>
  <a
    class="c1"
    href="#target8"
    target="_self"
  >
    Link 8
  </a>
  <a
    class="c1"
    href="#target9"
    target="_self"
  >
    Link 9
  </a>
  <a
    class="c1"
    href="#target10"
    target="_self"
  >
    Link 10
  </a>
</nav>
`;

exports[`<NavLinks/> should Match snapshot 1`] = `
[
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target1"
    target="_self"
  >
    Link 1
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target2"
    target="_self"
  >
    Link 2
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target3"
    target="_self"
  >
    Link 3
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target4"
    target="_self"
  >
    Link 4
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target5"
    target="_self"
  >
    Link 5
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target6"
    target="_self"
  >
    Link 6
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target7"
    target="_self"
  >
    Link 7
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target8"
    target="_self"
  >
    Link 8
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target9"
    target="_self"
  >
    Link 9
  </a>,
  .c0 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {

}

<a
    class="c0"
    href="#target10"
    target="_self"
  >
    Link 10
  </a>,
]
`;
